crates/circuit/src/circuit_instruction.rs

```diff
@@ -618,6 +622,51 @@ impl<'a, 'py> FromPyObject<'a, 'py> for OperationFromPython {
                     });
                 };
             }
+        } else if ob_name == "PauliProductMeasurement" {
```

Comment:
The gate name is camel case? I would have expected `"pauli_product_measurement"` or `"ppm"`.

Reply:
I thought that we should to use the camel case name because Pauli evolution gates are called `"PauliEvolution"`. But I agree, using snake case naming is more consistent, so it's renamed to `"pauli_product_measurement"` in [388e3d9](https://github.com/Qiskit/qiskit/pull/15126/commits/388e3d98025af0f85edfa4318cd6ce0fd3588ada).